shodan/host.go

```diff
@@ -60,12 +61,83 @@ func (v *HostVersion) String() string {
 	return string(*v)
 }
 
+// HostDHParams is the Diffie-Hellman parameters if available.
+type HostDHParams struct {
+	Prime       string       `json:"prime"`
+	PublicKey   string       `json:"public_key"`
+	Bits        int          `json:"bits"`
+	Generator   *HostVersion `json:"generator"`
```

Comment:
`HostVersion` should be renamed to something else. The reason for not using straight int is that shodan sometimes replies with weird stuff like string - "000000000000000000000000000000000002".